ring_doorbell/auth.py

```diff
@@ -47,44 +83,95 @@ def __fetch_token(self, username, password,
         :type auth_code: str
         """
         if auth_code:
-            headers = {}
+            headers = OAuth.HEADERS.copy()
             headers['2fa-support'] = 'true'
             headers['2fa-code'] = auth_code
 
-            return self._oauth.fetch_token(
+            token = self._oauth.fetch_token(
                 OAuth.ENDPOINT,
                 username=username,
                 password=password,
                 scope=OAuth.SCOPE,
                 headers=headers)
 
-        return self._oauth.fetch_token(
+            self.__default_token_updater(token)
+
+            return token
+
+        token = self._oauth.fetch_token(
             OAuth.ENDPOINT,
             username=username,
             password=password,
-            scope=OAuth.SCOPE)
+            scope=OAuth.SCOPE,
+            headers=OAuth.HEADERS)
+
+        self.__default_token_updater(token)
+
+        return token
 
-    # Python 2 doesn't support typing for return value,
-    #  removed: -> Dict[str, Union[str, int]]:
     def refresh_tokens(self):
         """Refreshes the auth tokens"""
-        token = self._oauth.refresh_token(OAuth.ENDPOINT)
+        token = self._oauth.refresh_token(
+            OAuth.ENDPOINT, headers=OAuth.HEADERS)
 
         if self.token_updater is not None:
             self.token_updater(token)
 
+        self.__default_token_updater(token)
         return token
 
-    # Python 2 doesn't support typing for return value,
-    #  removed: -> Response:
-    def request(self, method, resource, **kwargs):
-        """Does an http request, if token is expired, then it will refresh
-        :type method: str
-        :type resource: str
-        """
-        try:
-            return self._oauth.request(method, resource, **kwargs)
+    def query(self,
+              url,
+              attempts=RETRY_TOKEN,
+              method='GET',
+              raw=False,
+              extra_params=None,
+              json=None,
+              timeout=None):
+        """Query data from Ring API."""
+        # Configure timeout specific to this query
+        query_timeout = timeout or TIMEOUT
+
+        response = None
+        loop = 0
+        while loop <= attempts:
```

Comment:
I think that we should drop the loop. We should just call it once, if token is expired, once more and that's it.  Everything more is just hammering their API for no good reason.